lib/v2/screens/app/app.dart

```diff
@@ -89,24 +89,24 @@ class _AppState extends State<App> with WidgetsBindingObserver {
   Widget build(BuildContext context) {
     _navigatorKey = NavigationService.of(context).appNavigatorKey;
     return BlocProvider(
-      create: (context) => AppBloc(),
-      child: BlocListener<AppBloc, AppState>(
-        listenWhen: (previous, current) {
-          return current.pageCommand != null;
-        },
-        listener: (buildContext, state) {
-          var pageCommand = state.pageCommand;
-          BlocProvider.of<AppBloc>(buildContext).add(ClearPageCommand());
-          if (pageCommand is BottomBarNavigateToIndex) {
-            _pageController.jumpToPage(pageCommand.index);
-          } else if (pageCommand is ShowStopGuardianRecoveryFailed) {
-            SnackBarInfo(title: pageCommand.message, context: buildContext).show(buildContext);
-          } else if (pageCommand is ShowStopGuardianRecoverySuccess) {
-            SnackBarInfo(title: pageCommand.message, context: buildContext).show(buildContext);
-          }
-        },
-        child: Scaffold(
-          body: BlocBuilder<AppBloc, AppState>(builder: (context, state) {
+      create: (_) => AppBloc(),
+      child: Scaffold(
+        body: BlocConsumer<AppBloc, AppState>(
+          listenWhen: (_, current) => current.pageCommand != null,
+          listener: (context, state) {
+            var pageCommand = state.pageCommand;
+            BlocProvider.of<AppBloc>(context).add(ClearAppPageCommand());
+            if (pageCommand is BottomBarNavigateToIndex) {
+              _pageController.jumpToPage(pageCommand.index);
```

Comment:
I feel it is more natural to access from `state.index` instead of `state.pageCommand.index` but it is only a preference and of course when it is only one variable.